supplemental_notebooks/2020-07-30_pathway_analysis.Rmd

```diff
@@ -0,0 +1,702 @@
+  # Drop the column that contains the Ensembl IDs
+  dplyr::select(-gene_id) %>%
+  # Make the first column the gene symbols
+  dplyr::select(gene_symbol, dplyr::everything())
+```
+
+How many of the gene symbols are duplicated?
+
+```{r}
+sum(duplicated(norm_counts_df$gene_symbol))
+```
+
+For duplicated gene identifiers, average the values.
+
+```{r}
+collapsed_norm_counts_df <- norm_counts_df %>%
+  # For each gene symbol
+  dplyr::group_by(gene_symbol) %>%
+  # Take the mean value
+  dplyr::summarize_all(mean)
+```
+
+### Create files that can be used with GSEA or GenePattern
+
+We tried to get `fgsea::fgseaLabel()` to work for these data and received an uninformative error!
+
+```{r
+fgseaLabel(
+  pathway = hs_hallmark_list,  # list format instead of tidy format
+  mat = collapsed_norm_mat,   # matrix where rownames are gene symbols
+  labels = as.integer(as.factor(metadata_df$Group)),  # subtype as labels
+  nperm = 1000  # number of permutations
+)
+```
+
+Which produced:
+
+```
+Error in if (NR == 0) { : missing value where TRUE/FALSE needed
+```
+
+Uninformative!
+
+So instead, we'll show you how to create the files you'd need to run GSEA on [GenePattern](https://www.genepattern.org/).
+
+We'll need the expression data in [GCT (Gene Cluster Text) format](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GCT:_Gene_Cluster_Text_file_format_.28.2A.gct.29) and the phenotype labels for two group in the [CLS format](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29).
+
+We can use some helpful functions from the `ArrayTools` package to write these files out.
+First, we need to get the expression data (normalized counts) into a special object called an `ExpressionSet` and then we can write out a GCT file.
+
+```{r}
+# Collapsed normalized counts as a matrix
+collapsed_norm_mat <- collapsed_norm_counts_df %>%
+  tibble::column_to_rownames("gene_symbol") %>%
+  as.matrix()
+
+# Create an ExpressionSet
+pdx_expression_set <- ExpressionSet(
+  assayData = collapsed_norm_mat
+)
+
+# GCT file
+ArrayTools::output.gct(pdx_expression_set,
+                       # We don't need the file extension, the function
+                       # will add it for us
+                       filename = file.path(pdx_dir,
+                                            "pdx_medulloblastoma_norm_count"))
+```
+
+And now for our CLS file.
+
+```{r}
+# We need the sample identifiers as rownames
+phenotype_df <- metadata_df %>%
+  dplyr::select(Run, Group, treatment) %>%
+  tibble::column_to_rownames("Run")
+
+# CLS file
+ArrayTools::output.cls(phenotype_df,
+                       variable = "Group",
+                       filename = file.path(pdx_dir,
+                                            "pdx_medulloblastoma_molecular_subtype"))
+```
+
+Here's a screenshot of how we could use these files on GenePattern:
```

Comment:
Do you want to say anything about the options you chose here? I noticed that you are using Hallmark again...
```suggestion
Here's a screenshot of how we could use these files on [GenePattern](https://www.genepattern.org/):
```

Reply:
Yes, can do.